src/Microsoft.DotNet.Build.Tasks/GenerateTestExecutionScripts.cs

```diff
@@ -90,7 +90,7 @@ private void WriteShExecutionScript(string shExecutionTemplate, string outputPat
             foreach (string runCommand in TestCommands)
             {
                 testRunCommands.Append($"{runCommand}\n");
-                testRunEchoes.Append($"echo {runCommand}\n".Replace("(", "").Replace(")", ""));  // Remove parentheses from echo command to avoid errors on Linux
+                testRunEchoes.Append($"echo \"{runCommand}\"\n".Replace("(", "").Replace(")", ""));  // Remove parentheses from echo command to avoid errors on Linux
```

Comment:
what happens when runCommand contains a quote character itself?  I think to go this route you'd need to remove them before wrapping.